inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -630,18 +630,25 @@ std::vector<PointSpeedPair> InLaneCruisingPlugin::maneuvers_to_points(const std:
         } else {
           downsampled_points = carma_utils::containers::downsample_vector(centerline, config_.default_downsample_ratio);
         }
+
+        if (downsampled_centerline.size() != 0 && downsampled_points.size() != 0 // If this is not the first lanelet and the points are closer than 1m drop the first point to prevent overlap
+        && lanelet::geometry::distance2d(downsampled_points.front(), downsampled_centerline.back()) < 1.2) { 
+          ROS_DEBUG_STREAM("Dropping first point due to overlap");
+          downsampled_points = lanelet::BasicLineString2d(downsampled_points.begin() + 1, downsampled_points.end());
```

Comment:
Will begin + 1 here result in a bad alloc if the size is 1?

Reply:
I don't believe so. In the case of an std vector with size 1. vec.begin() + 1 == vec.end()

This code runs fine
```
    vector<int> vec = {0};
    vector<int> vec2 = vector<int>(vec.begin() + 1, vec.end());
    cout << "vec size: " << vec.size() << endl;
    cout << "vec2 size: " << vec2.size() << endl;
 ``` 
 
 However in the case of begin() + 2 you do get a bad_alloc. Since this